### PR TITLE
Update dataset tags to allow editing/addition of tags

### PR DESCRIPTION
**Pull request by davidsharp7**
### Problem

Currently there is functionality to add tags at both a column and dataset level, however, there is no way of maintaining or adding tags unless you go through the API.

Closes: #2741 

###
Solution

Update DatasetTags to include a split button menu and a new dialog/reducer for adding new tags.

Screenshot 2024-03-01 at 10 36 16
Screenshot 2024-03-01 at 10 37 56
Screenshot 2024-03-01 at 10 38 09

One-line summary:

Add Tag dialog box

### Checklist

- [x] You've [signed-off](https://github.com/MarquezProject/marquez/blob/main/CONTRIBUTING.md#sign-your-work) your work
- [x] Your changes are accompanied by tests (_if relevant_)
- [x] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [x] You've updated any relevant documentation (_if relevant_)
- [x] You've included a one-line summary of your change for the [`CHANGELOG.md`](https://github.com/MarquezProject/marquez/blob/main/CHANGELOG.md#unreleased) (_Depending on the change, this may not be necessary_).
- [x] You've versioned your `.sql` database schema migration according to [Flyway's naming convention](https://flywaydb.org/documentation/concepts/migrations#naming) (_if relevant_)
- [x] You've included a [header](https://github.com/MarquezProject/marquez/blob/main/CONTRIBUTING.md#copyright--license) in any source code files (_if relevant_)

### CODE REVIEW

File: web/src/components/datasets/DatasetTags.tsx [hunk 1]
```diff
@@ -112,22 +168,81 @@ const DatasetTags: React.FC<IProps> = (props) => {
...
         <MQText subheading>{i18next.t('dataset_tags.tags')}</MQText>
         {formatTags(datasetTags, tagData)}
-        <MQTooltip placement='top' title={i18next.t('dataset_tags.tooltip')} key='tag-tooltip'>
-          <IconButton
-            onClick={openDialog}
+        <ButtonGroup
+          variant='contained'
+          ref={anchorRef}
+          aria-label='Tags Nested Menu'
+          sx={{ height: '30px', width: '20px', marginLeft: '5px' }}
```
Comment (phixMe) [inline web/src/components/datasets/DatasetTags.tsx:186]: Try to use units of four for these kinds of operations. I believe if you say `1` or `2` that does the multiplication for you.

File: web/src/components/datasets/DatasetTags.tsx [hunk 2]
```diff
@@ -112,22 +168,81 @@ const DatasetTags: React.FC<IProps> = (props) => {
 
   return (
     <>
+      <Snackbar
+        open={snackbarOpen}
+        autoHideDuration={500}
+        style={{ zIndex: 9999 }}
```
Comment (phixMe) [inline web/src/components/datasets/DatasetTags.tsx:174]: This should be available in the theme.

File: web/src/store/reducers/tags.ts [hunk 3]
```diff
@@ -1,26 +1,33 @@
...
 
-type ITagsAction = ReturnType<typeof fetchTagsSuccess>
+type ITagsAction = ReturnType<typeof fetchTagsSuccess> & ReturnType<typeof addTagsSuccess>
 
 export default (state: ITagsState = initialState, action: ITagsAction): ITagsState => {
   const { type, payload } = action
   switch (type) {
     case FETCH_TAGS:
-      return { ...state, isLoading: true }
+      return { ...state }
```
Comment (phixMe) [inline web/src/store/reducers/tags.ts:24]: I think we wish to have the loading state between fetching and success.

**Review by phixMe — COMMENTED** [review]
Minor comments, Can you include a screenshot or video of the behavior for users to see the experience?

**Review by wslulciuc — APPROVED** [review]
LGTM! 💯

**Review by phixMe — APPROVED** [review]
Looks great!

**Pull request merged**